docs/working_with_data_sources.md

```diff
@@ -0,0 +1,222 @@
+
+```ruby
+# encoding: utf-8
+
+require 'gooddata'
+
+snowflake_ds_model = {
+          'dataSource' => {
+              'name' => 'My Data Source',
+              'alias' => 'ds_alias',
+              'prefix' => 'ds_prefix_',
+              'connectionInfo' => {
+                  'snowflake' => {
+                      'url' => 'jdbc:snowflake://test',
+                      'authentication' => {
+                          'basic' => {
+                              'userName' => 'testing_user',
+                              'password' => '***'
+                          }
+                      },
+                      'database' => 'db_test',
+                      'schema' => 'schema_test',
+                      'warehouse' => 'wh_test'
+                  }
+              }
+          }
+      }
+client = GoodData.connect
+snowflake_ds = client.create(GoodData::DataSource, snowflake_ds_model)
+snowflake_ds.save
+
+# After the data source is created then you can start working with it
+# Using Snowflake data source as source input in process
+project_test = client.projects('project_pid_testing')
+process_test = project_test.deploy_process({
+          name: 'Testing Process',
+          type: 'LCM',
+          component: {
+              name: 'gdc-cm-bricks',
+              version: '1'
+          },
+          dataSources:[{id: "#{snowflake_ds.id}"}]
+      })
+# Using schema, database and password of Snowflake data source as value of schedule parameters
+schedule_test = process_test.create_schedule(
+          '0 15 10 3 *',
+          'Testing Schedule',
+          params: {
+              username: 'test_us',
+              database: '${ds_alias.database}',
+              schema: '${ds_alias.schema}'
+          },
+          hidden_params: {
+              db_pass: '${ds_alias.password}'
+          }
+      )
+```
+
+Or you can create Amazon Redshift data source:
+
+```ruby
+# encoding: utf-8
+
+require 'gooddata'
+
+# Create Amazon Redshift data source with basic authentication
+redshift_ds_model = {
+            'dataSource' => {
+                'name' => 'My Data Source',
+                'alias' => 'ds_alias',
+                'prefix' => 'ds_prefix_',
+                'connectionInfo' => {
+                    'redshift' => {
+                        'url' => 'jdbc:redshift://test',
+                        'authentication' => {
+                            'basic' => {
+                                'userName' => 'testing_user',
+                                'password' => '***'
+                            }
+                        },
+                        'database' => 'db_test',
+                        'schema' => 'schema_test'
+                    }
+                }
+            }
+        }
+client = GoodData.connect
+redshift_ds = client.create(GoodData::DataSource, redshift_ds_model)
+redshift_ds.save
+
+# Create Amazon Redshift data source with access key and secret key authentication
```

Comment:
Creating an Amazon Redshift Data Source with access key and secret key authentication:

Reply:
Done fix it and no need ":" at the end